zrml/prediction-markets/src/migrations.rs

```diff
@@ -15,7 +15,246 @@
+            market_ids_per_dispute_iterator.collect();
+        total_weight = total_weight
+            .saturating_add(T::DbWeight::get().reads(market_ids_per_dispute.len() as u64));
+        for (dispute_start_block, market_ids) in &market_ids_per_dispute {
+            total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
+            let dispute_duration = T::DisputePeriod::get();
+
+            let new_dispute_start_block = dispute_start_block.saturating_add(dispute_duration);
+            MarketIdsPerDisputeBlock::<T>::insert(new_dispute_start_block, market_ids);
+        }
+        for (dispute_start_block, _market_ids) in market_ids_per_dispute {
+            MarketIdsPerDisputeBlock::<T>::remove(dispute_start_block);
+            total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
+        }
+
+        let market_ids_per_report_iterator: IterType<T> =
+            frame_support::migration::storage_key_iter::<_, _, Twox64Concat>(
+                b"PredictionMarkets",
+                b"MarketIdsPerReportBlock",
+            );
+
+        let market_ids_per_report: Vec<DisputeBlockToMarketIdsTuple<T>> =
+            market_ids_per_report_iterator.collect();
+        total_weight = total_weight
+            .saturating_add(T::DbWeight::get().reads(market_ids_per_report.len() as u64));
+        for (dispute_start_block, market_ids) in &market_ids_per_report {
+            total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
+            let dispute_duration = T::DisputePeriod::get();
+
+            let new_dispute_start_block = dispute_start_block.saturating_add(dispute_duration);
+            MarketIdsPerReportBlock::<T>::insert(new_dispute_start_block, market_ids);
+        }
+        for (dispute_start_block, _market_ids) in market_ids_per_report {
+            MarketIdsPerReportBlock::<T>::remove(dispute_start_block);
+            total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
+        }
+        StorageVersion::new(PREDICTION_MARKETS_NEXT_STORAGE_VERSION_FOR_MIGRATE_MARKET_IDS_STORAGE)
+            .put::<Pallet<T>>();
+        total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
+
+        log::info!("Completed storage migration of MigrateMarketIdsPerBlockStorage");
+        total_weight
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn pre_upgrade() -> Result<(), &'static str> {
+        Ok(())
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn post_upgrade() -> Result<(), &'static str> {
+        for (key, market_ids) in MarketIdsPerDisputeBlock::<T>::iter() {
+            for market_id in market_ids {
+                let market =
+                    T::MarketCommons::market(&market_id).map_err(|_| "invalid market_id")?;
+                let disputes = Disputes::<T>::get(market_id);
+                let dispute = disputes.last().ok_or("No dispute found")?;
```

Comment:
Maybe log the market id here. I think we should never have some item in `MarketIdsPerDisputeBlock` which has no last dispute.